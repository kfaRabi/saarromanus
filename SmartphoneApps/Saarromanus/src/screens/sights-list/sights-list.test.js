import 'react-native';
import React from 'react';
import { shallow } from 'enzyme';
import configureStore from 'redux-mock-store';

import SightsListScreen from './index';

const mockStore = configureStore();

const mockFn = jest.fn();
const navigation = {
	navigate: mockFn,
};

const initialState = {
	config: { checkForUpdate: false },
	sights: [
		{
			id: 'AK1AIRDavlLPkNjV1VDt',
			en: {
				name: 'Mithra chapel on Halberg',
				information: 'The worship of the sun...',
			},
			de: {
				name: 'Mithraskapelle am Halberg',
				information: 'Die Anbetung der Sonne ist ein Kult',
			},
			fr: {
				name: 'Chapelle de Mithras Ã  Halberg',
				information: "L'adoration du soleil est un culte",
			},
			thumbnail: 'https://firebasestorage.googleapis.com/v0/b',
			resourceName: 'front_large.jpg',
			sight_category: 'qXG2ZqlHWLQiDLYea1qn',
		},
	],
};

let wrapper;
let store;
let component;

beforeAll(() => {
	store = mockStore(initialState);
	wrapper = shallow(
		<SightsListScreen store={store} navigation={navigation} />
	).dive();
	component = wrapper.dive();
});

describe('<SightsListScreen />', () => {
	it('renders properly', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('has one main <ScrollView /> component', () => {
		expect(component.find(`[data-test-id='mainContainer']`).length).toBe(1);
	});

	describe('Sights List', () => {
		it('should be an array of sights', () => {
			expect(Array.isArray(wrapper.props().sights)).toBe(true);
		});
		it('length should be equal to the number of sights passed via the store', () => {
			expect(wrapper.props().sights.length).toBe(
				initialState.sights.length
			);
		});
	});
});
