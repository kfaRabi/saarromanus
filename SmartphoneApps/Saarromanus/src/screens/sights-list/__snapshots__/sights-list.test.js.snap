// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SightsListScreen /> renders properly 1`] = `
<SightsListScreen
  checkUpdateStatus={false}
  getSightsByCategory={[Function]}
  navigation={
    Object {
      "navigate": [MockFunction],
    }
  }
  populateSightsByCategory={[Function]}
  sights={
    Array [
      Object {
        "de": Object {
          "information": "Die Anbetung der Sonne ist ein Kult",
          "name": "Mithraskapelle am Halberg",
        },
        "en": Object {
          "information": "The worship of the sun...",
          "name": "Mithra chapel on Halberg",
        },
        "fr": Object {
          "information": "L'adoration du soleil est un culte",
          "name": "Chapelle de Mithras Ã  Halberg",
        },
        "id": "AK1AIRDavlLPkNjV1VDt",
        "resourceName": "front_large.jpg",
        "sight_category": "qXG2ZqlHWLQiDLYea1qn",
        "thumbnail": "https://firebasestorage.googleapis.com/v0/b",
      },
    ]
  }
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
/>
`;
